File: app/src/main/java/g/takeru/renshu/architecture/mvp/LoginPresenter.java
package g.takeru.renshu.architecture.mvp;

import g.takeru.renshu.kotlin.User;

/**
 * Created by takeru on 2017/10/11.
 */

public class LoginPresenter extends NullCheckPresenter<LoginView>{

    public void doLoginTask(String name, String password) {
        getView().loginSuccess(new User("takeru", 20));

//        ApiManager.Companion.getInstance().getApi()
//                .getData()
//                .subscribeOn(Schedulers.newThread())
//                .observeOn(AndroidSchedulers.mainThread())
//                .subscribe(
//                        response ->{
//                            Timber.d("success: " + response);
//                        },
//                        throwable -> {
//                            Timber.d("failure: " + throwable.getMessage());
//                        }
//                        );
    }
}

File: app/src/main/java/g/takeru/renshu/kotlin/architecture/mvp/LoginPresenter.kt
package g.takeru.renshu.kotlin.architecture.mvp

import g.takeru.renshu.kotlin.User

/**
 * Created by takeru on 2017/10/11.
 */

class LoginPresenter : BasePresenter<LoginView> {

    private var view: LoginView? = null

    override fun attachView(view: LoginView) {
        this.view = view
    }

    override fun detachView() {
        view = null
    }

    fun doLoginTask(name: String, password: String) {
        view?.loginSuccess(User("takeru", 20))
    }
}
